### PR TITLE
FEAT: impl rolling.corr

### LINKED ISSUE #300: FEAT: implement `xorbits.pandas.window.Rolling.corr`
### Is your feature request related to a problem? Please describe
Calculate rolling corr is a frequent operation in stock analysis. We should implement it.

### LINKED ISSUE #303: BUG: multi-indexed dataframes' shape
### Describe the bug
The shape of a multi-indexed dataframe is incorrect.

### To Reproduce
In Xorbits,
```python
df = pd.DataFrame({"foo": (1, 2, 3), "bar": (4, 5, 6)})
c = df.rolling(window=2).agg('corr')
c.shape
```
you got `(3, 2)`.

But in pandas, you got `(6, 2)`.

### PR DESCRIPTION
**Pull request by UranusSeven**
## What do these changes do?
Implement `DataFrame.rolling.corr`.

## Related issue number
Resolves #300 
Resolves #303 

## Check code requirements

- [x] tests added / passed (if needed)
- [ ] Ensure all linting tests pass

### CODE REVIEW

**Comment by aresnow1** [comment]
Could you rebase main branch, the coverage configuration has been modified.

**Comment by UranusSeven** [comment]
> Could you rebase main branch, the coverage configuration has been modified.

Sure.

File: python/xorbits/_mars/dataframe/window/rolling/core.py [hunk 1]
```diff
@@ -184,6 +184,11 @@ def skew(self, **kwargs):
     def kurt(self, **kwargs):
         return self.aggregate("kurt", **kwargs)
 
+    def corr(self, **kwargs):
+        # for convenience, since the calculation of corr is axis irrelevant.
+        self._axis = 0
```
Comment (aresnow1) [inline python/xorbits/_mars/dataframe/window/rolling/core.py:189]: Better to raise an error if use axis is not 0.

File: python/xorbits/_mars/dataframe/window/rolling/aggregation.py [hunk 2]
```diff
@@ -243,7 +243,8 @@ def _find_extra_chunks_for_int_window(cls, op, inp, cur_chunk_index):
 
         # find prev chunks
         i = cur_chunk_index[axis]
-        rest = window if not center else window // 2
+        # handle all the possibilities of op.closed.
+        rest = window if not center else window // 2 + 1
```
Comment (aresnow1) [inline python/xorbits/_mars/dataframe/window/rolling/aggregation.py:247]: Have you added a test to cover this corner case?

File: python/xorbits/_mars/dataframe/window/rolling/aggregation.py [hunk 3]
```diff
@@ -432,13 +440,19 @@ def tile(cls, op):
         if out.ndim == 1:
             params["shape"] = (inp.shape[0],)
         else:
-            params["shape"] = (inp.shape[0], params["shape"][1])
+            if op.func == "corr" and op.func_kwargs.get("pairwise", True):
```
Comment (aresnow1) [inline python/xorbits/_mars/dataframe/window/rolling/aggregation.py:443]: This line is same as line 407, we can check at beginning and record with a variable.

File: python/xorbits/_mars/dataframe/window/rolling/aggregation.py (hunk 2 above)
Comment (UranusSeven) [reply python/xorbits/_mars/dataframe/window/rolling/aggregation.py:247]: Yes, these lines are covered when op.closed being set to values other than 'right'.

File: python/xorbits/_mars/dataframe/window/rolling/core.py (hunk 1 above)
Comment (UranusSeven) [reply python/xorbits/_mars/dataframe/window/rolling/core.py:189]: `axis=1` is also valid.

But when the aggregation function is `corr`, `axis 0` and `axis 1` produce same results.

File: python/xorbits/_mars/dataframe/window/rolling/aggregation.py (hunk 3 above)
Comment (UranusSeven) [reply python/xorbits/_mars/dataframe/window/rolling/aggregation.py]: Done.

**Review by aresnow1 — APPROVED** [review]
LGTM

**Review by ChengjieLi28 — APPROVED** [review]
